Sem. 03/Pract. 03/mySolutions/Task02.cpp

```diff
@@ -0,0 +1,329 @@
+}
+
+bool isPerfectOffer(const JobOffer& offer, const int maxCoworkers, const int minVacancyDays, const long long minSalary)
+{
+	return offer.coworkers <= maxCoworkers && offer.numberOfVacancyDays >= minVacancyDays && offer.salary >= minSalary;
+}
+
+void perfectOffer(const char* filePath, const int maxCoworkers, const int minVacancyDays, const long long minSalary)
+{
+	std::ifstream in(ALL_OFFERS_BINARY_FILE_PATH, std::ios::binary);
+	std::ofstream out(PERFECT_OFFERS_BINARY_FILE_PATH, std::ios::app, std::ios::binary);
+
+	if (!in.is_open())
+	{
+		std::cerr << "Couldn't find file to read from!";
+		return;
+	}
+
+	if (!out.is_open())
+	{
+		std::cerr << "Couldn't find file to write into!";
+		return;
+	}
+
+	while (!in.eof())
+	{
+		JobOffer offer = readJobOfferFromFile(in);
+
+		if (isPerfectOffer(offer, maxCoworkers, minVacancyDays, minSalary) && !existOffer(PERFECT_OFFERS_BINARY_FILE_PATH, offer.companyName))
+			saveJobOffer(out, offer);
+	}
+
+	in.close();
+	out.close();
+}
+
+void printAllJobOffersFromFile(const char* filePath)
+{
+	std::ifstream in(filePath, std::ios::binary);
+
+	if (!in.is_open())
+	{
+		std::cerr << "Couldn't find fie!";
+		return;
+	}
+
+	size_t fileSize = getFileSize(in);
+	size_t numberOfOffers = fileSize / sizeof(JobOffer);
+	JobOffer* offers = new JobOffer[numberOfOffers];
+
+	in.read((char*)offers, fileSize);
+
+	for (size_t i = 0; i < numberOfOffers; i++)
+		printJobOffer(offers[i]);
+
+	in.close();
+}
+
+void addJobOffer()
+{
+	std::ofstream out(ALL_OFFERS_BINARY_FILE_PATH, std::ios::app | std::ios::binary);
+
+	if (!out.is_open())
+	{
+		std::cerr << "Couldn't find file to add to!";
+		return;
+	}
+
+	JobOffer offer;
+	readJobOfferFromConsole(offer);
+	saveJobOffer(out, offer);
+
+	out.close();
+}
+
+void checkIfCompanyExists()
+{
+	char companyName[26];
+
+	std::cout << "Enter company name: ";
+	std::cin >> companyName;
+	std::cout << existOffer(ALL_OFFERS_BINARY_FILE_PATH, companyName) << std::endl;
+}
+
+void runApp()
+{
+	bool endApp = false;
+	while (!endApp)
+	{
+		std::cout << "Enter 'a' to add offer;" << std::endl
+			<< "Enter 'i' to print all offers;" << std::endl
+			<< "Enter 's' to check if job offer exists;" << std::endl
+			<< "Enter 'f' to print all offers with salary above certain number;" << std::endl
+			<< "Enter 'q' to exit." << std::endl
+			<< "Enter a key: ";
+
+		char key;
+		std::cin >> key;
+		switch (key)
+		{
+		case 'a':
+			std::cout << "Add offer" << std::endl;
+			addJobOffer();
+			break;
+		case 'i':
+			std::cout << "Print:" << std::endl;
+			printAllJobOffersFromFile(ALL_OFFERS_BINARY_FILE_PATH);
+			break;
+		case 's':
+			std::cout << "Check if company exists" << std::endl;
+			checkIfCompanyExists();
+			break;
+		case 'f':
+			long long minSalary;
+			std::cout << "Print all offers with salary more than: ";
+			std::cin >> minSalary;
+			filterBySalaryOffers(ALL_OFFERS_BINARY_FILE_PATH, minSalary);
+			break;
+		case 'q':
+			endApp = true;
+			break;
+		}
```

Comment:
add default case for wrong command

Reply:
Should I repair my code and send it again or leave it like this?

Reply:
You don't have to send it but if you me to have a look, you can reopen this pull request